crates/toml_config/src/lib.rs

```diff
@@ -94,7 +95,7 @@ pub mod __private {
 
                     let value = crate::toml::ser::to_string(self).unwrap();
 
-                    output.push_str(&format!("{} = {}\n", field_name, value));
+                    output.push_str(&format!("{} = {}\n\n", field_name, value));
```

Comment:
Hmm, how to you feel about some options to configure this?

For me it lacks readability if everything sticks together and I do have a big config file.

So

```toml
# This is a comment
a = "somevalue"
# do something blabla
b = "someothervalue"
```

For me is way harder to read than
```toml
# This is a comment
a = "somevalue"

# do something blabla
b = "someothervalue"
```

But I do have some more problems aswell:

```toml

[api]
# The address to listen on
# This will be an HTTPS server
address = "127.0.0.1"

# The port to listen on
# TCP
port = 8443


```

So right now the first line is empty (I don't like that at all) and two empty lines in the end I do not like either (yes I know the second line is from my changes)

What do you think? If I would make some kind of optional config how would you want it to look like?

Reply:
If you’re able to implement it, you could try checking for a lack of preceding top-level variables, which I think is the cause of the top newline. The same goes for configuration: if you are able to implement choosing between single and double newlines in a somewhat clean way, it is reasonable to give that option. I won’t be able to spend time trying to do so myself, however. Also, a word of warning is that the current proc macro is somewhat duct tape-like. But parametering the config trait over a Formatting struct or similar might be workable.